// @bsv/sdk/dist/cjs/src/transaction/http/FetchHttpClient.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FetchHttpClient = void 0;
/**
 * Adapter for Node Https module to be used as HttpClient
 */
class FetchHttpClient {
    constructor(fetch) {
        this.fetch = fetch;
    }
    async request(url, options) {
        var _a;
        const fetchOptions = {
            method: options.method,
            headers: options.headers,
            body: JSON.stringify(options.data)
        };
        const res = await this.fetch(url, fetchOptions);
        const mediaType = res.headers.get('Content-Type');
        const data = ((_a = mediaType === null || mediaType === void 0 ? void 0 : mediaType.startsWith('application/json')) !== null && _a !== void 0 ? _a : false)
            ? await res.json()
            : await res.text();
        return {
            ok: res.ok,
            status: res.status,
            statusText: res.statusText,
            data: data
        };
    }
}
exports.FetchHttpClient = FetchHttpClient;
